lib/rubocop/cop/rspec/aggregate_examples.rb

```diff
@@ -0,0 +1,294 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks if example groups contain more than one aggregateable example.
+      #
+      # @example
+      #
+      #   # bad
+      #   describe do
+      #     specify do
+      #       expect(number).to be_positive
+      #       expect(number).to be_odd
+      #     end
+      #
+      #     it { is_expected.to be_prime }
+      #   end
+      #
+      #   # good
+      #   describe do
+      #     specify do
+      #       expect(number).to be_positive
+      #       expect(number).to be_odd
+      #       is_expected.to be_prime
+      #     end
+      #   end
+      #
+      #   # fair - subject has side effects
+      #   describe do
+      #     specify do
+      #       expect(multiply_by(2)).to be_multiple_of(2)
+      #     end
+      #
+      #     specify do
+      #       expect(multiply_by(3)).to be_multiple_of(3)
+      #     end
+      #   end
+      #
+      #   # The following example will fail if aggregated due to the side
+      #   # effects of the `validate_presence_of` matcher as it leaves an empty
+      #   # comment after itself on the subject making it invalid and the
+      #   # subsequent expectation to fail.
+      #
+      #   # bad, but should not be automatically correctable
+      #   describe do
+      #     it { is_expected.to validate_presence_of(:comment) }
+      #     it { is_expected.to be_valid }
+      #   end
+      #
+      #   # Block expectation syntax is deliberately not supported due to:
+      #   # 1. `subject { -> { ... } }` syntax being hard to detect
+      #   # E.g.:
+      #   it { is_expected.to do_something }
+      #   # looks like an example with non-block syntax, but it might be
+      #   # depending on how the subject is defined. If the subject is defined
+      #   # in a `shared_context`, it's impossible to detect that at all.
+      #   #
+      #   # 2. Aggregation should use composition with an `.and`. Also,
+      #   # aggregation of the `not_to` expectations is barely possible when a
+      #   # matcher doesn't provide a negated variant.
+      #   #
+      #   # 3. Aggregation of block syntax with non-block syntax should be in a
+      #   # specific order.
+      #
+      class AggregateExamples < Cop # rubocop:disable Metrics/ClassLength
+        include RangeHelp
+
+        MSG = 'Aggregate with the example above.'.freeze
+        MSG_FOR_EXPECTATIONS_WITH_SIDE_EFFECTS =
+          "#{MSG} IMPORTANT! Pay attention to the expectation order, some " \
+          'of the matchers have side effects.'.freeze
+
+        MATCHERS_WITH_SIDE_EFFECTS = %w[
```

Comment:
Better move this to config, so users can add their own

Reply:
Agree. Also, this setting may eventually be also used by another cop that detects existing examples that have expectations with side effects.